tests/trainer/test_trainer_eval.py

```diff
@@ -140,7 +140,7 @@ def test_eval_hparams(composer_trainer_hparams: TrainerHparams):
     composer_trainer_hparams.evaluators = [
         EvaluatorHparams(
             label="eval1",
-            eval_interval='3ep',  # should NEVER run, since we train just 2 epochs
+            eval_interval='3ep',  # will run, since eval_at_fit_end = True
```

Comment:
Can you add a seperate test case for this, ideally using the non-hparams codepath? Makes sense that this change is needed, since this functionality changes the default behavior that hparams uses, but would be nice to have a seperate test for that flag specifically (perhaps parameterize the option)

Reply:
Yup can do!

Reply:
Done!